src/Datadog.Trace/Agent/Api.cs

```diff
@@ -41,7 +41,7 @@ public async Task<bool> SendTracesAsync(Span[][] traces)
             var retryCount = 1;
             var sleepDuration = 100; // in milliseconds
 
-            Log.Debug("Sending {Count} traces to the DD agent", traces.Length);
+            Log.Debug<int>("Sending {Count} traces to the DD agent", traces.Length);
```

Comment:
Shouldn't the `<int>` be added implicitly by type inference? Or is that because of the other overloads?

Reply:
Yeah, unfortunately in a few cases it's not possible due to the overloads. Without it, this call is ambiguous between the following:

```csharp
void Information<T>(string messageTemplate, T property);
void Information(string messageTemplate, [CallerLineNumber] int sourceLine = 0, [CallerFilePath] string sourceFile = "");
```

These cases are pretty rare, but are unavoidable if we use `[CallerLineNumber]` etc as the RFC requires